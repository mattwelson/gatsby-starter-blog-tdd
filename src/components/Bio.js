import React from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'
import Image from 'gatsby-image'

import { rhythm } from '../utils/typography'

export const PureBio = ({ author, social, avatar }) => (
  <div
    style={{
      display: `flex`,
      marginBottom: rhythm(2.5),
    }}
  >
    <Image
      fixed={avatar.childImageSharp.fixed}
      alt={author}
      style={{
        marginRight: rhythm(1 / 2),
        marginBottom: 0,
        minWidth: 50,
        borderRadius: `100%`,
      }}
    />
    <p>
      Written by <strong>{author}</strong> who lives and works in Minneapolis
      building silly things.
      {` `}
      <a href={`https://twitter.com/${social.twitter}`}>
        You should follow him on Twitter
      </a>
    </p>
  </div>
)

PureBio.propTypes = {
  author: PropTypes.string.isRequired,
  social: PropTypes.shape({
    twitter: PropTypes.string.isRequired,
  }).isRequired,
  avatar: PropTypes.shape({
    childImageSharp: PropTypes.shape({
      fixed: PropTypes.shape(),
    }),
  }),
}

function Bio() {
  return (
    <StaticQuery
      query={bioQuery}
      render={data => {
        const { author, social } = data.site.siteMetadata
        return <PureBio author={author} social={social} avatar={data.avatar} />
      }}
    />
  )
}

const bioQuery = graphql`
  query BioQuery {
    avatar: file(absolutePath: { regex: "/profile-pic.jpg/" }) {
      childImageSharp {
        fixed(width: 50, height: 50) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    site {
      siteMetadata {
        author
        social {
          twitter
        }
      }
    }
  }
`

export default Bio
