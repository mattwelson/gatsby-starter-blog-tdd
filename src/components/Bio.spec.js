import React from 'react'
import renderer from 'react-test-renderer'

import { PureBio } from './Bio'

describe('PureBio', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PureBio
          author="Test Author"
          social={{ twitter: 'testtwitter.com' }}
          avatar={{
            childImageSharp: { fixed: '' },
          }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
