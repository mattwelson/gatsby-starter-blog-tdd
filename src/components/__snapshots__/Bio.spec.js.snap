// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PureBio renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginBottom": "4.375rem",
    }
  }
>
  <img
    alt="Test Author"
    style={
      Object {
        "borderRadius": "100%",
        "marginBottom": 0,
        "marginRight": "0.875rem",
        "minWidth": 50,
      }
    }
  />
  <p>
    Written by 
    <strong>
      Test Author
    </strong>
     who lives and works in Minneapolis building silly things.
     
    <a
      href="https://twitter.com/testtwitter.com"
    >
      You should follow him on Twitter
    </a>
  </p>
</div>
`;
